File: 347-Top-K-Frequent-Elements.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number[]}
 */
var topKFrequent = function (nums, k) {
  const freq = {}
  const sorted = []
  const result = []

  nums.forEach(num => freq[num] = (freq[num] || 0) + 1)

  for (const entry in freq) {
    sorted.push([entry, freq[entry]])
  }

  sorted.sort((a, b) => b[1] - a[1])

  return sorted.slice(0, k).map(entry => entry[0])
};

File: 347-Top-K-Frequent-Elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        priority_queue<pair<int, int>> pq;
        unordered_map<int, int> mp;
        vector<int> ans;

        for (int num : nums)
            mp[num]++;

        for (auto entry : mp)
            pq.push({entry.second, entry.first});

        while (k--) {
            ans.push_back(pq.top().second);
            pq.pop();
        }

        return (ans);
    }
};